### PR TITLE
Replace REST API with GraphQL

### PR DESCRIPTION
**Pull request by guillem-gelabert**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (anwyss) [inline .vscode/launch.json:4]: maybe these comments can be removed

File: src/users/users.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+export class UsersResolver {
+  constructor(
+    private usersService: UsersService,
+    private authService: AuthService,
+  ) {}
+
+  @Query()
+  @UseGuards(JwtAuthGuard)
+  async whoAmI(@CurrentUser() user: User): Promise<User> {
+    const currentUser = await this.usersService.findOne(user);
```
Comment (anwyss) [inline src/users/users.resolver.ts]: should this await be in a `try` block?

File: src/users/users.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    if (!currentUser) throw new Error('Authentication Error');
+    return currentUser;
+  }
+
+  @Mutation()
+  async createUser(@Args('input') input: CreateUserDto): Promise<AuthSuccess> {
+    const user = await this.usersService.create(input);
```
Comment (anwyss) [inline src/users/users.resolver.ts]: please put the user asignment in a `try` block

File: src/users/users.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    let accessToken = '';
+    if (user) {
+      accessToken = await this.authService.signToken(user.username);
```
Comment (anwyss) [inline src/users/users.resolver.ts]: `try` please

File: src/users/users.service.ts [hunk 5]
```diff
@@ -12,11 +13,25 @@ export class UsersService {
...
+    } catch (e) {
+      throw new Error(e);
+    }
   }
 
-  async findOne(username: CreateUserDto['username']): Promise<User> {
-    return this.userModel.findOne({ username }).lean();
+  async findOne(user: Partial<CreateUserDto>): Promise<User | null> {
+    try {
+      return this.userModel.findOne({ username: user.username }).lean();
```
Comment (anwyss) [inline src/users/users.service.ts]: should this be awaited?

File: src/users/users.resolver.ts (hunk 4 above)
Comment (guillem-gelabert) [reply src/users/users.resolver.ts]: you mean `try` harder 😄

File: src/users/users.service.ts (hunk 5 above)
Comment (guillem-gelabert) [reply src/users/users.service.ts]: All async functions return automatically a Promise, as I don't need to process the value before I return I can directly forward the Promise returned by `findOne`, whoever wants to look inside will have to `await` it.

Comment (guillem-gelabert) [reply src/users/users.service.ts]: I'll remove the `try/catch` as this Promise cannot be rejected

**Pull request merged**